frontend/src/hooks/useSmartTrainerInterface.ts

```diff
@@ -49,61 +140,158 @@ export const useSmartTrainerInterface = (): SmartTrainer => {
-    );
-    indoorBikeDataCharacteristic &&
-      setIndoorBikeDataCharacteristic(indoorBikeDataCharacteristic);
+      const indoorBikeDataCharacteristic =
+        await fitnessMachineService?.getCharacteristic('indoor_bike_data');
 
-    setIsConnected(true);
-    logEvent('smart trainer connected');
-  };
-  const disconnect = async () => {
-    if (indoorBikeDataCharacteristic) {
-      await indoorBikeDataCharacteristic.stopNotifications();
-      console.log('> Indoor bike data notifications stopped');
-      indoorBikeDataCharacteristic.removeEventListener(
+      if (!indoorBikeDataCharacteristic) {
+        return dispatch({
+          type: 'set_error',
+          errorMessage: 'Could not connect to indoor bike data characteristic.',
+        });
+      }
+
+      await indoorBikeDataCharacteristic?.startNotifications();
+      indoorBikeDataCharacteristic?.addEventListener(
         'characteristicvaluechanged',
         handleIndoorBikeDataUpdate
       );
-      setIndoorBikeDataCharacteristic(null);
+
+      dispatch({
+        type: 'set_fitness_machine_characteristics',
+        device,
+        indoorBikeDataCharacteristic,
+        fitnessMachineControlPointCharacteristic,
+      });
+    } catch (e) {
+      logEvent(
+        'Could not connect to device as Fitness Machine. Trying to connect to cycling power.'
+      );
+      const cyclingPowerService = await server?.getPrimaryService(
+        'cycling_power'
+      );
+      if (!cyclingPowerService) {
+        return dispatch({
+          type: 'set_error',
+          errorMessage: 'Could not connect to cycling power service.',
+        });
+      }
+
+      const cyclingPowerMeasurementCharacteristic =
+        await cyclingPowerService.getCharacteristic(
+          'cycling_power_measurement'
+        );
+
+      if (!cyclingPowerMeasurementCharacteristic) {
+        return dispatch({
+          type: 'set_error',
+          errorMessage:
+            'Could not connect to cycling power measurement characteristic.',
+        });
+      }
+
+      await cyclingPowerMeasurementCharacteristic?.startNotifications();
+      cyclingPowerMeasurementCharacteristic?.addEventListener(
+        'characteristicvaluechanged',
+        handleCyclingPowerUpdate
+      );
+
+      dispatch({
+        type: 'set_cycling_power_characteristics',
+        device,
+        cyclingPowerMeasurementCharacteristic,
+      });
+
+      toast({
+        title:
+          'Could not connect to your smart trainer as a fitness machine. ' +
+          'This means that Dundring will only be able to read the power ' +
+          'data, but not set resistance. If you have a smart trainer ' +
+          'that should be capable of this, please ensure that you have ' +
+          'the latest firmware update, and try again. If that does not ' +
+          'work, please contact us at our Slack workspace.',
+        isClosable: true,
+        duration: 20000,
+        status: 'warning',
+      });
```

Comment:
I'm also planning to implement a button on the profile page that will disable this toast. Per now you will get it every time you connect with a non-fitness-machine, which might be annoying.